switch jinja Undefined with AnsibleUndefinedVariable

TrayserCassa opened the pull request:
Fixes #63129

##### SUMMARY
I looked up at the ansible.error classes and switched it with the jinja.Undefined 

Fixes #63129

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/vars/hostvars.py

##### ADDITIONAL INFORMATION

Commit:
switch jinja Undefined with AnsibleUndefinedVariable

issue 63129

mkrizek commented on lib/ansible/vars/hostvars.py:
```diff
@@ -75,7 +74,7 @@ def raw_get(self, host_name):
         host = self._find_host(host_name)
         if host is None:
-            return Undefined(name="hostvars['%s']" % host_name)
+            return AnsibleUndefinedVariable(name="hostvars['%s']" % host_name)
```
`AnsibleUndefinedVariable` is an exception class. You probably want `AnsibleUndefined` which is Ansible's subclass of Jinja2's `Undefined`. See https://github.com/ansible/ansible/blob/bdd0fac606265f92e6521540106accf19a1ae4d0/lib/ansible/template/__init__.py#L232-L246

TrayserCassa replied:
yes your right, also the error msg is better. Changed it.

Commit:
Using AnsibleUndefined instead of AnsibleUndefinedVariable

Commit:
Using name has better error output then key

name:
fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'foo' is undefined\n\nThe error appears to be in '/home/patrick/dev/python/ansible_test/install.yml': line 4, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n  tasks:\n    - debug:\n      ^ here\n"}

key:
fatal: [localhost]: FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'ansible.vars.hostvars.HostVars object' has no attribute 'foo'\n\nThe error appears to be in '/home/patrick/dev/python/ansible_test/install.yml': line 4, column 7, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n  tasks:\n    - debug:\n      ^ here\n"}

Commit:
check for instance in __getitem__ use now AnsibleUndefined

samdoran commented:
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, please add unit tests.

Since this is changing a pretty fundamental change, please add `ci_complete` to your commit message in order to run all tests.

Commit:
added changelog file

Commit:
remove space after file suffix

Commit:
small change to trigger to run all tests ci_complete

mkrizek commented:
Superseded by https://github.com/ansible/ansible/pull/64959
